packages/light-apps/src/IdentityCard/IdentityCard.tsx

```diff
@@ -120,39 +87,57 @@ export class IdentityCard extends React.PureComponent<Props, State> {
     }
   }
 
-  private handleToggleApp = () => {
+  getAddress = () => {
+    return this.props.location.pathname.split('/')[2];
+  }
+
+  getButtonText = () => {
+    const currentLocation = location.pathname.split('/')[1].toLowerCase();
+
+    const to = currentLocation === 'identity' ? 'transfer' : 'identity';
+    return stringUpperFirst(to);
+  }
+
+  getName = () => {
+    const { keyring } = this.context;
+    const address = this.getAddress();
+
+    return address && keyring.getAccount(address).getMeta().name;
+  }
+
+  handleToggleApp = () => {
     const { location, history } = this.props;
     const address = this.getAddress();
     const currentLocation = location.pathname.split('/')[1].toLowerCase();
 
     const to = currentLocation === 'identity' ? 'transfer' : 'identity';
-    const buttonText = stringUpperFirst(to);
 
-    const nextLocation = {
-      pathname: `/${to}/${address}`,
-      state: {
-        buttonText: buttonText
-      }
-    };
-
-    history.push(nextLocation);
+    history.push(`/${to}/${address}`);
   }
 
-  private onChangePassword = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
+  onChangePassword = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Btw YJ I removed the privates. I was kind of confusing: we were sometimes using it, sometimes not. We could:
- either remove everywhere (which I did here)
- or if you prefer, we could put it everywhere, that's fine by me too. I just prefer when things are consistent.

as for method order, I'm following +/- [this](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md#rule-details). I added "+/-" because `onChangePassword = () => {...}` is actually a property, but we can consider these ones as methods, and exceptions (like the one you put) are also allowed.

Reply:
yes, fair enough, considering the component event handlers are implied to be private.

I haven't seen any other "standard" for component method ordering - thanks for the link!

Would you mind adding this to the README? We have been following a rule of thumb that subcomponent render methods come below the main `render()`, but that link seems to prefer the main `render()` at the very bottom. Could we make that an exception as well - for no reason other than that it's what's been consistently followed already.

Reply:
Done! I added in the CONTRIBUTING, as it's only relevant for people contributing. If it gets too long we could separate it into a STYLEGUIDE.md